soci/gzip_indexer.go

```diff
@@ -0,0 +1,147 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package soci
+
+// #cgo CFLAGS: -I${SRCDIR}/../c/
+// #cgo LDFLAGS: -L${SRCDIR}/../out -lindexer -lz
+// #include "indexer.h"
+// #include <stdlib.h>
+// #include <stdint.h>
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+type GzipIndexer struct {
```

Comment:
Let's rename this to 'GzipIndex'. 'Indexer' sounds like a struct whose sole purpose it to generate the index, but this a just a wrapper for the C struct.

Reply:
Done.